tests/core/test_model.py

```diff
@@ -4450,3 +4468,77 @@ def test_incremental_by_partition(sushi_context, assert_exp_eq):
             """
         )
         load_sql_based_model(expressions)
+
+
+@pytest.mark.parametrize(
+    ["model_def", "path", "expected_name"],
+    [
+        [
+            """dialect duckdb,""",
+            """models/test_schema/test_model.sql,""",
+            "test_schema.test_model",
+        ],
+        [
+            """dialect duckdb,""",
+            """models/test_model.sql,""",
+            "test_model",
+        ],
+        [
+            """dialect duckdb,""",
+            """models/inventory/db/test_schema/test_model.sql,""",
+            "db.test_schema.test_model",
+        ],
+        ["""name test_model,""", """models/schema/test_model.sql,""", "test_model"],
+    ],
+)
+def test_model_table_name_inference(
+    sushi_context: Context, model_def: str, path: str, expected_name: str
+):
+    model = load_sql_based_model(
+        d.parse(
+            f"""
+        MODEL (
+            {model_def}
+        );
+        SELECT a FROM tbl;
+        """,
+            default_dialect="duckdb",
+        ),
+        path=Path(f"$root/{path}"),
+        infer_names=True,
+    )
+    assert model.name == expected_name
+
+
+@pytest.mark.parametrize(
+    ["path", "expected_name"],
+    [
+        [
+            """models/test_schema/test_model.py""",
+            "test_schema.test_model",
+        ],
+        [
+            """models/inventory/db/test_schema/test_model.py""",
+            "db.test_schema.test_model",
+        ],
+    ],
+)
+def test_python_model_name_inference(tmp_path: Path, path: str, expected_name: str) -> None:
+    init_example_project(tmp_path, dialect="duckdb")
+    config = Config(
+        default_connection=DuckDBConnectionConfig(),
```

Comment:
This is the [default value](https://github.com/TobikoData/sqlmesh/blob/main/sqlmesh/core/config/root.py#L81) for `default_connection` so no need to set it here:
```suggestion
```